### ISSUE TITLE
new doc tags and doc popup optimization

### ISSUE THREAD

**Issue by ghost** [EmmyLua/IntelliJ-EmmyLua#154]
Here a some of the documentation tags JavaDoc uses. I'd like to have them available for Lua as well.

- [ ] `{@code}`   **=** _`{@code sum = x + y}`_
- [ ] `{@docRoot}` **=** `<a href="_{@docRoot}_/copyright.html">`
- [ ] `{@link class#member  label}` **=** Use the _`{@link #getComponentAt(int, int) getComponentAt}`_ method > Use the getComponentAt method.
- [ ] `{@linkplain class#member  label}` **=** Refer to _`{@linkplain add() the overridden method}`_ > Refer to the overridden method.
- [x] `@since since-text` **=** `@since` 1.2
- [x] `@author name-text` **=** `@author`  John Doe
- [x] `@deprecated deprecated-text` **=** `@deprecated`  As of JDK 1.1, replaced by _`{@link #setBounds(int,int,int,int)}`_
- [x] `@version version-text` **=** `@version` 2018.2
- [ ] HTML tags **=** `<a href="">, <p>, <li>, <i>, <b>, <h4>, <pre>` etc.
- [ ] `@apiNote`, `@implSpec` and `@implNote` **=** _`@implNote`_ Some long text.

The html tags can include pretty much any of the existing tags. 

- [x] Java, JavaScript and TypeScript are now using the optimized and redesigned documentation popups. [Here an example of how it looks like](https://blog.jetbrains.com/webstorm/2018/02/webstorm-2018-1-eap-181-3494/) I wonder if that's possible Lua to have too? Thank you in advance!

**Comment by dsmgit** [comment]
Any progress made?

**Comment by tangzx** [comment]
Yes but too slow, These days I am busy on working. Show you some finished features
![image](https://user-images.githubusercontent.com/3354360/42510305-11745ce2-8481-11e8-8f06-e7c8c7cdcf07.png)

**Comment by dsmgit** [comment]
Looking nice already!
